### ISSUE TITLE
设置不同的batchsize时会导致训练开始时卡住，没有任何错误信息被记录或者打印

### ISSUE THREAD

**Issue by BossCrab-jyj** [PaddlePaddle/PaddleSeg#3499]
### 问题确认 Search before asking

- [X] 我已经查询[历史issue](https://github.com/PaddlePaddle/PaddleSeg/issues)(包括open与closed)，没有发现相似的bug。I have searched the [open and closed issues](https://github.com/PaddlePaddle/PaddleSeg/issues) and found no similar bug report.

### Bug描述 Describe the Bug

训练代码：python -m paddle.distributed.launch tools/train.py        --config configs/quick_start/pp_liteseg_optic_disc_512x512_1k.yml        --do_eval        --use_vdl        --save_interval 1        --save_dir output --num_workers 12  --log_iters 1 --batch_size 12  可以正常训练。
训练代码：python -m paddle.distributed.launch tools/train.py        --config configs/quick_start/pp_liteseg_optic_disc_512x512_1k.yml        --do_eval        --use_vdl        --save_interval 1        --save_dir output --num_workers 60 --log_iters 1 --batch_size 60或者--num_workers 12  --num_workers 60 或者--num_workers 12  --batch_size 24卡住。

### 以下是运行信息：
------------------------------------------------
LAUNCH INFO 2023-09-12 11:08:21,790 -----------  Configuration  ----------------------
LAUNCH INFO 2023-09-12 11:08:21,790 auto_parallel_config: None
LAUNCH INFO 2023-09-12 11:08:21,790 devices: None
LAUNCH INFO 2023-09-12 11:08:21,790 elastic_level: -1
LAUNCH INFO 2023-09-12 11:08:21,790 elastic_timeout: 30
LAUNCH INFO 2023-09-12 11:08:21,790 gloo_port: 6767
LAUNCH INFO 2023-09-12 11:08:21,790 host: None
LAUNCH INFO 2023-09-12 11:08:21,790 ips: None
LAUNCH INFO 2023-09-12 11:08:21,790 job_id: default
LAUNCH INFO 2023-09-12 11:08:21,790 legacy: False
LAUNCH INFO 2023-09-12 11:08:21,790 log_dir: log
LAUNCH INFO 2023-09-12 11:08:21,790 log_level: INFO
LAUNCH INFO 2023-09-12 11:08:21,790 log_overwrite: False
LAUNCH INFO 2023-09-12 11:08:21,790 master: None
LAUNCH INFO 2023-09-12 11:08:21,790 max_restart: 3
LAUNCH INFO 2023-09-12 11:08:21,790 nnodes: 1
LAUNCH INFO 2023-09-12 11:08:21,790 nproc_per_node: None
LAUNCH INFO 2023-09-12 11:08:21,791 rank: -1
LAUNCH INFO 2023-09-12 11:08:21,791 run_mode: collective
LAUNCH INFO 2023-09-12 11:08:21,791 server_num: None
LAUNCH INFO 2023-09-12 11:08:21,791 servers:
LAUNCH INFO 2023-09-12 11:08:21,791 start_port: 6070
LAUNCH INFO 2023-09-12 11:08:21,791 trainer_num: None
LAUNCH INFO 2023-09-12 11:08:21,791 trainers:
LAUNCH INFO 2023-09-12 11:08:21,791 training_script: tools/train.py
LAUNCH INFO 2023-09-12 11:08:21,791 training_script_args: ['--config', 'configs/quick_start/pp_liteseg_optic_disc_512x512_1k.yml', '--do_eval', '--use_vdl', '--save_interval', '1', '--save_dir', 'output', '--num_workers', '60', '--log_iters', '1', '--batch_size', '60']
LAUNCH INFO 2023-09-12 11:08:21,791 with_gloo: 1
LAUNCH INFO 2023-09-12 11:08:21,791 --------------------------------------------------
LAUNCH INFO 2023-09-12 11:08:21,792 Job: default, mode collective, replicas 1[1:1], elastic False
LAUNCH INFO 2023-09-12 11:08:21,793 Run Pod: wbgglu, replicas 6, status ready
LAUNCH INFO 2023-09-12 11:08:21,883 Watching Pod: wbgglu, replicas 6, status running
======================= Modified FLAGS detected =======================
FLAGS(name='FLAGS_selected_gpus', current_value='0', default_value='')
=======================================================================
I0912 11:08:23.754560 2598655 tcp_utils.cc:181] The server starts to listen on IP_ANY:41136
I0912 11:08:23.754935 2598655 tcp_utils.cc:130] Successfully connected to 127.0.1.1:41136
W0912 11:08:25.743657 2598655 gpu_resources.cc:119] Please NOTE: device: 0, GPU Compute Capability: 8.9, Driver API Version: 12.2, Runtime API Version: 11.7
W0912 11:08:25.744632 2598655 gpu_resources.cc:149] device: 0, cuDNN Version: 8.9.
2023-09-12 11:08:27 [WARNING]   Add the `num_classes` in train_dataset and val_dataset config to model config. We suggest you manually set `num_classes` in model config.
2023-09-12 11:08:28 [INFO]
------------Environment Information-------------
platform: Linux-5.15.0-82-generic-x86_64-with-glibc2.29
Python: 3.8.10 (default, May 26 2023, 14:05:08) [GCC 9.4.0]
Paddle compiled with cuda: True
NVCC: Not Available
cudnn: 8.9
GPUs used: 6
CUDA_VISIBLE_DEVICES: None
GPU: ['GPU 0: NVIDIA GeForce', 'GPU 1: NVIDIA GeForce', 'GPU 2: NVIDIA GeForce', 'GPU 3: NVIDIA GeForce', 'GPU 4: NVIDIA GeForce', 'GPU 5: NVIDIA GeForce']
GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
PaddleSeg: 2.8.0
PaddlePaddle: 2.5.1
OpenCV: 4.5.5
------------------------------------------------
2023-09-12 11:08:28 [INFO]
---------------Config Information---------------
batch_size: 60
iters: 1000
train_dataset:
  dataset_root: /home/jh/jyj/datasets/my_data
  mode: train
  num_classes: 2
  train_path: /home/jh/jyj/datasets/my_data/train_list.txt
  transforms:
  - max_scale_factor: 2.0
    min_scale_factor: 0.5
    scale_step_size: 0.25
    type: ResizeStepScaling
  - crop_size:
    - 512
    - 512
    type: RandomPaddingCrop
  - type: RandomHorizontalFlip
  - brightness_range: 0.5
    contrast_range: 0.5
    saturation_range: 0.5
    type: RandomDistort
  - type: Normalize
  type: Dataset
val_dataset:
  dataset_root: /home/jh/jyj/datasets/my_data
  mode: val
  num_classes: 2
  transforms:
  - type: Normalize
  type: Dataset
  val_path: /home/jh/jyj/datasets/my_data/val_list.txt
optimizer:
  momentum: 0.9
  type: SGD
  weight_decay: 4.0e-05
lr_scheduler:
  end_lr: 0
  learning_rate: 0.01
  power: 0.9
  type: PolynomialDecay
loss:
  coef:
  - 1
  - 1
  - 1
  types:
  - type: CrossEntropyLoss
  - type: CrossEntropyLoss
  - type: CrossEntropyLoss
model:
  backbone:
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/PP_STDCNet2.tar.gz
    type: STDC2
  num_classes: 2
  type: PPLiteSeg
------------------------------------------------

2023-09-12 11:08:28 [INFO]      Set device: gpu
2023-09-12 11:08:28 [INFO]      Use the following config to build model
model:
  backbone:
    pretrained: https://bj.bcebos.com/paddleseg/dygraph/PP_STDCNet2.tar.gz
    type: STDC2
  num_classes: 2
  type: PPLiteSeg
2023-09-12 11:08:28 [INFO]      Loading pretrained model from https://bj.bcebos.com/paddleseg/dygraph/PP_STDCNet2.tar.gz
2023-09-12 11:08:28 [INFO]      There are 265/265 variables loaded into STDCNet.
2023-09-12 11:08:28 [INFO]      Convert bn to sync_bn
2023-09-12 11:08:28 [INFO]      Use the following config to build train_dataset
train_dataset:
  dataset_root: /home/jh/jyj/datasets/my_data
  mode: train
  num_classes: 2
  train_path: /home/jh/jyj/datasets/my_data/train_list.txt
  transforms:
  - max_scale_factor: 2.0
    min_scale_factor: 0.5
    scale_step_size: 0.25
    type: ResizeStepScaling
  - crop_size:
    - 512
    - 512
    type: RandomPaddingCrop
  - type: RandomHorizontalFlip
  - brightness_range: 0.5
    contrast_range: 0.5
    saturation_range: 0.5
    type: RandomDistort
  - type: Normalize
  type: Dataset
2023-09-12 11:08:28 [INFO]      Use the following config to build val_dataset
val_dataset:
  dataset_root: /home/jh/jyj/datasets/my_data
  mode: val
  num_classes: 2
  transforms:
  - type: Normalize
  type: Dataset
  val_path: /home/jh/jyj/datasets/my_data/val_list.txt
2023-09-12 11:08:28 [INFO]      If the type is SGD and momentum in optimizer config, the type is changed to Momentum.
2023-09-12 11:08:28 [INFO]      Use the following config to build optimizer
optimizer:
  momentum: 0.9
  type: Momentum
  weight_decay: 4.0e-05
2023-09-12 11:08:28 [INFO]      Use the following config to build loss
loss:
  coef:
  - 1
  - 1
  - 1
  types:
  - type: CrossEntropyLoss
  - type: CrossEntropyLoss
  - type: CrossEntropyLoss
[2023-09-12 11:08:28,762] [    INFO] distributed_strategy.py:160 - distributed strategy initialized
[2023-09-12 11:08:28,763] [ WARNING] fleet.py:290 - The dygraph parallel environment has been initialized.
[2023-09-12 11:08:29,915] [    INFO] topology.py:266 - Total 1 data comm group(s) create successfully!
[2023-09-12 11:08:29,929] [    INFO] topology.py:266 - Total 6 model comm group(s) create successfully!
[2023-09-12 11:08:29,942] [    INFO] topology.py:266 - Total 6 pipe comm group(s) create successfully!
[2023-09-12 11:08:29,956] [    INFO] topology.py:266 - Total 6 sharding comm group(s) create successfully!
[2023-09-12 11:08:29,969] [    INFO] topology.py:217 - HybridParallelInfo: rank_id: 0, mp_degree: 1, sharding_degree: 1, pp_degree: 1, dp_degree: 6, mp_group: [0],  sharding_group: [0], pp_group: [0], dp_group: [0, 1, 2, 3, 4, 5], check/clip group: [0]

程序一直卡在这里不动，单块GPU显存显示只占用 3492MiB / 24564MiB，GPU利用率为0，使用htop查看cpu一直在运行，连续观察3-5分钟，依旧卡住不动。

### 复现环境 Environment

platform: Linux-5.15.0-82-generic-x86_64-with-glibc2.29
Python: 3.8.10 (default, May 26 2023, 14:05:08) [GCC 9.4.0]
Paddle compiled with cuda: True
NVCC: Not Available
cudnn: 8.9
GPUs used: 6
CUDA_VISIBLE_DEVICES: None
GPU: ['GPU 0: NVIDIA GeForce', 'GPU 1: NVIDIA GeForce', 'GPU 2: NVIDIA GeForce', 'GPU 3: NVIDIA GeForce', 'GPU 4: NVIDIA GeForce', 'GPU 5: NVIDIA GeForce']，6张4090，显存约150gb
cpu：64核
GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
PaddleSeg: 2.8.0
PaddlePaddle: 2.5.1
OpenCV: 4.5.5

### Bug描述确认 Bug description confirmation

- [X] 我确认已经提供了Bug复现步骤、代码改动说明、以及环境信息，确认问题是可以复现的。I confirm that the bug replication steps, code change instructions, and environment information have been provided, and the problem can be reproduced.

### 是否愿意提交PR？ Are you willing to submit a PR?

- [ ] 我愿意提交PR！I'd like to help by submitting a PR!

**Comment by Asthestarsfalll** [comment]
似乎应该给Paddle仓库提issue
